### ISSUE TITLE
long boot time

### ISSUE THREAD

**Issue by danielwuensche** [Askannz/optimus-manager#490]
currently I experience long boot times. 

Manjaro with both 5.16 and 5.17 kernel
Intel integrated graphics
Nvidia RTX 2070 max-q

systemd-analyze blame  
```31.625s optimus-manager.service```

journalctl -u optimus-manager  
```-- Boot 84d37f902fe44f63bd48f95f3461fec8 --  
Apr 15 11:12:30 daniel-fusion15 systemd[1]: Starting Optimus Manager Commands Daemon...  
Apr 15 11:12:31 daniel-fusion15 python3[713]: [87] INFO: # Daemon pre-start hook  
Apr 15 11:12:31 daniel-fusion15 python3[713]: [88] INFO: Removing /etc/X11/xorg.conf.d/10-optimus-manager.conf (if present)
Apr 15 11:12:31 daniel-fusion15 python3[713]: [89] INFO: Copying /etc/optimus-manager/optimus-manager.conf to /var/lib/optimus-manager/tmp/config_copy.conf
Apr 15 11:12:31 daniel-fusion15 python3[713]: [102] INFO: Startup mode is: auto
Apr 15 11:12:31 daniel-fusion15 python3[713]: [124] INFO: Effective startup mode is: integrated
Apr 15 11:12:31 daniel-fusion15 python3[713]: [125] INFO: Writing state {'type': 'pending_pre_xorg_start', 'requested_mode': 'integrated', 'current_mode': None}
Apr 15 11:12:31 daniel-fusion15 python3[713]: [126] INFO: Daemon pre-start hook completed successfully.
Apr 15 11:12:31 daniel-fusion15 python3[713]: [126] INFO: Calling Xorg pre-start hook.
Apr 15 11:12:31 daniel-fusion15 python3[784]: [71] INFO: # Xorg pre-start hook
Apr 15 11:12:31 daniel-fusion15 python3[784]: [71] INFO: Previous state was: {'type': 'pending_pre_xorg_start', 'requested_mode': 'integrated', 'current_mode': None}
Apr 15 11:12:31 daniel-fusion15 python3[784]: [72] INFO: Requested mode is: integrated
Apr 15 11:12:31 daniel-fusion15 python3[909]: modinfo: ERROR: Module bbswitch not found.
Apr 15 11:12:31 daniel-fusion15 python3[915]: modinfo: ERROR: Module acpi_call not found.
Apr 15 11:12:33 daniel-fusion15 python3[784]: [2388] INFO: Available modules: ['nouveau', 'nvidia', 'nvidia_drm', 'nvidia_modeset', 'nvidia_uvm']
Apr 15 11:12:33 daniel-fusion15 python3[784]: [2388] INFO: Unloading modules ['nvidia_drm', 'nvidia_modeset', 'nvidia_uvm', 'nvidia'] (if loaded)
Apr 15 11:12:33 daniel-fusion15 python3[784]: [2402] INFO: switching=none, nothing to do
Apr 15 11:13:02 daniel-fusion15 python3[784]: [30887] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
Apr 15 11:13:02 daniel-fusion15 python3[784]: [30887] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20220415T111231', 'requested_mode': 'integrated'}
Apr 15 11:13:02 daniel-fusion15 python3[784]: [30888] INFO: Xorg pre-start hook completed successfully.
Apr 15 11:13:02 daniel-fusion15 systemd[1]: Started Optimus Manager Commands Daemon.
Apr 15 11:13:02 daniel-fusion15 python3[1150]: [6] INFO: # Commands daemon
Apr 15 11:13:02 daniel-fusion15 python3[1150]: [6] INFO: Opening UNIX socket
Apr 15 11:13:02 daniel-fusion15 python3[1150]: [6] INFO: Awaiting commands
```
it hangs after `INFO: switching=none, nothing to do`

In `dmesg -H` I see quite a few messages regarding power management
`pci 0000:38:00.0: can't change power state from D3cold to D0 (config space inaccessible)`

I normally use dynamic power management "fine", also tried "no"

I suppose optimus-manager tries to disable the dgpu but can't access it.
What can I do to further troubleshoot/fix the issue?

**Comment by danielwuensche** [comment]
Actually the dGPU now never turns off, the lowest I saw in nvidia-smi was 5W. 
All these issues started happening when I reinstalled grub - I'm dualbooting with Windows, reinstalled Windows which kicked grub out of the UEFI boot options. I don't see a reason why reinstalling grub should cause all that - all changes were made in /etc/default/grub and that file wasn't touched by the install-process.
